docs/tutorials/pi/make-a-plant-watering-robot.md

```diff
@@ -0,0 +1,469 @@
+Make sure to add your board (in this example, named `local`) to this component's `depends_on`, and `"max_rpm": 1000` and `"pwm": "12"` (the pin # that you wired the PWM to) to `attributes`:
+
+{{< tabs name="Configure an Pump Motor" >}}
+{{% tab name="Config Builder" %}}
+
+<img src="../../img/plant-watering-pi/pump-motor-config-builder.png" alt="Creation of a pump motor in the Viam app config builder." style="max-width:800px" />
+
+{{% /tab %}}
+{{% tab name="JSON Template" %}}
+
+```json {class="line-numbers linkable-line-numbers"}
+// Board JSON ... },
+{
+    "name": "water-pump",
+    "type": "motor",
+    "model": "gpio",
+    "attributes": {
+    "pins": {
+        "a": "",
+        "b": "",
+        "dir": "",
+        "pwm": "12"
+    },
+    "board": "local",
+    "max_rpm": 1000
+    },
+    "depends_on": [
+    "local"
+    ]
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### Configure the Capacitive Soil Moisture Sensor as a Custom Sensor Component Model
+
+As the Capacitive Sensor Moisture Sensor is not currently one of Viam's built-in [sensor component](/components/sensor/) models, you now must use the Viam Python SDK to configure this sensor as a [custom resource](/program/extend/sdk-as-server/).
+
+- In this case, the custom resource is a custom component model that extends the Viam [sensor class](https://python.viam.dev/autoapi/viam/components/sensor/sensor/index.html).
+- You instantiate this custom component as a server with the `viam.rpc.server` class.
+- Then, you add the component server as a remote part of your `plant-watering-robot`.
+
+If you haven't already, install the Viam Python SDK by following the instructions [here](https://python.viam.dev/).
+
+Navigate to the directory you created when [testing your soil moisture readings](#test-your-soil-moisture-readings-on-your-pi), `plant-watering-robot`, and create a new file, <file>sensor.py</file>.
+
+For example, run the following commands to create and open the file:
+
+``` shell
+cd plant-watering-robot
+touch sensor.py
+vim sensor.py
+```
+
+Paste the following code into <file>sensor.py</file>:
+
+```python
+import asyncio
+
+from viam.components.sensor import Sensor
+from viam.rpc.server import Server
+
+import busio
+import digitalio
+import board
+import adafruit_mcp3xxx.mcp3008 as MCP
+from adafruit_mcp3xxx.analog_in import AnalogIn
+import RPi.GPIO as GPIO
+
+class MoistureSensor(Sensor):
+
+   def __init__(self, name: str):
+        super(MoistureSensor,self).__init__(name)
+        sensor_pin = 4
+        GPIO.setmode(GPIO.BCM)
+        GPIO.setup(sensor_pin, GPIO.IN)
+
+
+    # Implement the Viam Sensor API's get_readings() method
+    async def get_readings(self):
+        x = 0
+        input = []
+
+        # create the spi bus
+        spi = busio.SPI(clock=board.SCK, MISO=board.MISO, MOSI=board.MOSI)
+
+        # create the cs (chip select)
+        cs = digitalio.DigitalInOut(board.D25) # TODO: Need to check this: example was D5. Documentation is confusing.
```

Comment:
This & the other place with TODO / `.D25` will be edited, I had to update Olivia's original code https://docs.google.com/document/d/16hZ98EsN5r9P1GU8gQDwVYXhyzv_kp5M/edit# to use a new library